frank/fit.py

```diff
@@ -203,71 +252,46 @@ def determine_geometry(u, v, vis, weights, inc, pa, dra, ddec, geometry_type,
-        Source right ascension offset from 0
-    ddec : float, unit = arcsec
-        Source declination offset from 0
-    geometry_type: string, from {'known', 'gaussian'}
-        Specifies how the geometry is determined. Options:
-            'known' : The user-provided geometry will be used
-            'gaussian' : Determine the geometry by fitting a Gaussian
-    fit_phase_offset: bool
-        Whether to fit for the source's right ascension offset and declination
-        offset from 0
+    model : dict
+        Dictionary containing model parameters the fit uses
 
     Returns
     -------
     geom : SourceGeometry object
         Fitted geometry (see frank.geometry.SourceGeometry)
     """
 
-    from frank import geometry
-
     logging.info('  Determining disc geometry')
 
-    if geometry_type == 'known':
+    if model['geometry']['type'] == 'known':
         logging.info('    Using your provided geometry for deprojection')
-        if all(x == 0 for x in (inc, pa, dra, ddec)):
-            logging.info("      N.B.: All geometry parameters are 0, so I won't"
-                         " apply any geometry correction to the visibilities")
-        geom = geometry.FixedGeometry(inc, pa, dra, ddec)
-
-    elif geometry_type == 'gaussian':
-        if fit_phase_offset:
-            logging.info('    Fitting Gaussian to determine geometry')
+        geom = geometry.FixedGeometry(model['geometry']['inc'],
+                                      model['geometry']['pa'],
+                                      model['geometry']['dra'],
+                                      model['geometry']['ddec']
+                                      )
+
+    elif model['geometry']['type'] == 'gaussian':
+        if model['geometry']['fit_phase_offset']:
             geom = geometry.FitGeometryGaussian()
 
         else:
-            logging.info('    Fitting Gaussian to determine geometry'
-                         ' (not fitting for phase center)')
-            geom = geometry.FitGeometryGaussian(phase_centre=(dra, ddec))
+            geom = geometry.FitGeometryGaussian(phase_centre=(model['geometry']['dra'],
+                                                              model['geometry']['ddec']))
 
-        t1 = time.time()
         geom.fit(u, v, vis, weights)
-        logging.info('    Time taken for geometry %.1f sec' %
-                     (time.time() - t1))
-    else:
-        raise ValueError("geometry_type must be one of 'known' or 'gaussian'")
 
-    logging.info('    Using: inc  = {:.2f} deg,\n           PA   = {:.2f} deg,\n'
-                 '           dRA  = {:.2e} mas,\n'
-                 '           dDec = {:.2e} mas'.format(geom.inc, geom.PA,
-                                                       geom.dRA*1e3,
-                                                       geom.dDec*1e3))
+    else:
+        raise ValueError("geometry_type in your parameter file must be one of"
+                         " 'known' or 'gaussian'")
 
```

Comment:
You may want to add this logging back in here

Reply:
I don't follow, I already have that `raise ValueError` in the current commit.

Reply:
You moved the logging printing the geometry to geometry.py, I've asked for it to be removed from there, so you might want to print the geometry in fit.py